### PR TITLE
#15640 adding the messages to report errors on parsing velocity on th…

### PR DESCRIPTION
**Pull request by jdotcms**
…e edit mode

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java [hunk 1]
```diff
@@ -77,5 +82,43 @@ public final Template getTemplate(IHTMLPage page, PageMode mode) {
...
+
+            systemMessageBuilder.setMessage(message.toString())
+                    .setLife(DateUtil.FIVE_SECOND_MILLIS)
+                    .setType(MessageType.SIMPLE_MESSAGE)
+                    .setSeverity(MessageSeverity.ERROR);
+
+            SystemMessageEventUtil.getInstance().
+                    pushMessage(systemMessageBuilder.create(), Arrays.asList(user.getUserId()));
+        } catch (Exception ex) {
+            Logger.error(this, ex.getMessage(), ex);
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java]: We should include the "name" in the error message, ideally the Identifier as well. Useful for Support.

File: dotCMS/src/main/java/com/dotcms/exception/ExceptionUtil.java [hunk 2]
```diff
@@ -374,8 +354,13 @@ public String getMessage() {
      * @return String
      */
     public static String getCurrentStackTraceAsString () {
-        final StringBuilder builder = new StringBuilder();
         final StackTraceElement [] traces = Thread.currentThread().getStackTrace();
+        return getStackTraceAsString(traces);
+    }
+
+    public static String getStackTraceAsString (final StackTraceElement [] traces) {
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/exception/ExceptionUtil.java]: Done

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java [hunk 3]
```diff
@@ -77,5 +82,43 @@ public final Template getTemplate(IHTMLPage page, PageMode mode) {
                 + page.getLanguageId() + "." + VelocityType.HTMLPAGE.fileExtension);
     }
 
+    protected void handleParseError (final ParseErrorException e, final String name, final User user) {
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java]: it is ok

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java (hunk 1 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java]: ok

**Review by jgambarios — APPROVED** [review]

**Pull request merged**